Tests/PackageLoadingTests/PD_5_3_LoadingTests.swift

```diff
@@ -177,29 +177,7 @@ class PackageDescription5_3LoadingTests: PackageDescriptionLoadingTests {
             let (_, validationDiagnostics) = try loadAndValidateManifest(content, observabilityScope: observability.topScope)
             XCTAssertNoDiagnostics(observability.diagnostics)
             testDiagnostics(validationDiagnostics) { result in
-                result.check(diagnostic: "invalid type for binary product 'FooLibrary'; products referencing only binary targets must have a type of 'library'", severity: .error)
-            }
-        }
-
-        do {
-            let content = """
-                import PackageDescription
-                let package = Package(
-                    name: "Foo",
-                    products: [
-                        .executable(name: "FooLibrary", targets: ["Foo"]),
-                    ],
-                    targets: [
-                        .binaryTarget(name: "Foo", path: "Foo.xcframework"),
-                    ]
-                )
-                """
-
-            let observability = ObservabilitySystem.makeForTesting()
-            let (_, validationDiagnostics) = try loadAndValidateManifest(content, observabilityScope: observability.topScope)
-            XCTAssertNoDiagnostics(observability.diagnostics)
-            testDiagnostics(validationDiagnostics) { result in
-                result.check(diagnostic: "invalid type for binary product 'FooLibrary'; products referencing only binary targets must have a type of 'library'", severity: .error)
```

Comment:
This diagnostic can longer be emitted for this particular case, because we cannot know whether a binary target is vending an executable or a library during manifest parsing since we need to crack open the referenced artifact to determine that. There will still be a diagnostic for this case, but it is handled by `PackageBuilder` and already covered by other existing tests.